test/python/test_ibmqjob.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name,missing-docstring,broad-except
+
+# Copyright 2017 IBM RESEARCH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+"""IBMQJob Test."""
+
+import unittest
+import time
+import numpy
+from scipy.stats import chi2_contingency
+
+from qiskit import (ClassicalRegister, QuantumCircuit, QuantumRegister,
+                    QuantumJob)
+import qiskit._compiler
+from qiskit.backends.ibmq import IBMQProvider
+from qiskit.backends.basejob import JobStatus
+from .common import requires_qe_access, QiskitTestCase, slow_test
+
+
+class TestIBMQJob(QiskitTestCase):
+    """
+    Test ibmqjob module.
+    """
+
+    @classmethod
+    @requires_qe_access
+    def setUpClass(cls, QE_TOKEN, QE_URL):
+        # pylint: disable=arguments-differ
+        super().setUpClass()
+        # create QuantumCircuit
+        qr = QuantumRegister(2, 'q')
+        cr = ClassicalRegister(2, 'c')
+        qc = QuantumCircuit(qr, cr)
+        qc.h(qr[0])
+        qc.cx(qr[0], qr[1])
+        qc.measure(qr, cr)
+        cls._qc = qc
+        cls._provider = IBMQProvider(QE_TOKEN, QE_URL)
+
+    def test_run_simulator(self):
+        backend = self._provider.get_backend('ibmqx_qasm_simulator')
+        qobj = qiskit._compiler.compile(self._qc, backend)
+        quantum_job = QuantumJob(qobj, backend, preformatted=True)
+        job = backend.run(quantum_job)
+        result = job.result()
+        counts_qx = result.get_counts(result.get_names()[0])
+        counts_ex = {'00': 512, '11': 512}
+        states = counts_qx.keys() | counts_ex.keys()
+        # contingency table
+        ctable = numpy.array([[counts_qx.get(key, 0) for key in states],
+                              [counts_ex.get(key, 0) for key in states]])
+        self.log.info('states: %s', str(states))
+        self.log.info('ctable: %s', str(ctable))
+        contingency = chi2_contingency(ctable)
+        self.log.info('chi2_contingency: %s', str(contingency))
+        self.assertGreater(contingency[1], 0.01)
+
+    @slow_test
+    def test_run_device(self):
+        backend = self._provider.get_backend('ibmqx4')
+        qobj = qiskit._compiler.compile(self._qc, backend)
+        quantum_job = QuantumJob(qobj, backend, preformatted=True)
+        job = backend.run(quantum_job)
+        while not job.done:
+            print(job.status)
```

Comment:
This should probably be `self.log.info` instead of `print`. It prints out a lot of extra lines during testing.